### DISCUSSION TITLE
Git action cant interact with git repository on private repo

### DISCUSSION THREAD

**Discussion by plenzjr** [community/community#25275] [Actions]
<p>I’m trying to setup an action that build python sphinx docs on every pull request.<br>
The repository is an organization private repository with pro account enabled.<br>
I’m not the owner.</p>
<p>but first, i created a personal repository and set up an environment for that and everything worked.</p>
<p>when i try to run same task on the private repository, i get the error:</p>
<pre><code class="lang-auto">Run git status
  git status  shell: /usr/bin/bash -e {0}
  env:
    DEFAULT_BRANCH: main
fatal: could not read Username for 'https://github.com': No such device or address
Error: Process completed with exit code 128.
</code></pre>
<p>here is the action:</p>
<pre data-code-wrap="yml"><code class="lang-nohighlight">name: Sphinx Pages

on: push
# on: pull_request

env:
  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}

jobs:
  main:
    name: Merge Main
    runs-on: ubuntu-latest
    steps:
      - name: git checkout
        uses: actions/checkout@v2
        with:
          persist-credentials: false
          fetch-depth: 0
          ref: ${{ github.head_ref }}

      - name: config git
        run: |
          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
          git config --local user.name "github-actions[bot]"
          git config pull.rebase false

      - name: check git
        run: git remote show origin -n

      - name: git status
        run: git status
      # errors happens here
</code></pre>
<p>my user can create a branch, pull, push e create pull request<br>
not sure if it’s about the user privileges.</p>
<p>as there are more devs creating pull request, should i create a secret for the action or the default <code>GITHUB_TOKEN</code> is enough for the task?</p>
<p>thank you in advance</p>

**Comment by jsoref** [comment]
<p>Why did you do this:</p>
<blockquote>
<pre><code>     persist-credentials: false</code></pre>
</blockquote>

**Comment by plenzjr** [comment]
<aside class="quote no-group" data-username="paulogil2010" data-post="1" data-topic="231853">
<div class="title">
<div class="quote-controls"></div>
 paulogil2010:</div>
<blockquote>
<p><code>actions/checkout@v2</code></p>
</blockquote>
</aside>
<p>just follow an example</p>

**Comment by jsoref** [comment, marked as answer]
<p>Well, that flag is what’s preventing you from doing what you want…</p>

**Comment by plenzjr** [comment]
<p>Thank you!<br>
it was exactly the problem!</p>
